plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryConnectorModule.java

```diff
@@ -44,48 +40,23 @@ public void configure(Binder binder)
     {
         // BigQuery related
         binder.bind(BigQueryReadClientFactory.class).in(Scopes.SINGLETON);
+        binder.bind(BigQueryClientFactory.class).in(Scopes.SINGLETON);
+
+        newOptionalBinder(binder, IdentityCacheMapping.class)
+                .setDefault()
+                .to(IdentityCacheMapping.SingletonIdentityCacheMapping.class)
+                .in(Scopes.SINGLETON);
```

Comment:
Add a comment why a singleton mapping is safe (and if/when we need to reconsider this).